### PR TITLE
STYLE, TYP Turn black formatter off for overloads

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Following on from the monthly meeting's discussion, some things were suggested:

- turning off the black formatter for the overloads, keeping them in `.py` files (what I've done here)
- moving overloads to `.pyi` files, writing a script to keep them in sync (I tried that this, and it seems that `mypy` behaves a bit differently with a `.pyi` file than it does with a `.py` file - will look into this more closely, but it'll take more time)
- keeping overloads out of `.py` files and "injecting" them before running `mypy` via a script - IMO this might be slightly disruptive to the usual workflow of just running `mypy pandas`, but other than that seems like a potentially good idea
- doing away with overloads, and just using `cast` and `assert` - TBH this strikes me as more error-prone, and if pandas will expose types to users, then I think it'd be better to have the correct type revealed rather than having them do `cast` / `assert`

Even if `inplace` is removed in version 2, there are other places in the codebase where there's a parameter which changes the return type, so I do think finding a solution here is worthwhile.

For now, I've gone with the first solution, though I will give some more thought to the second one

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
looks like `pep8speaks` doesn't respect `noqa`...

**Comment by MarcoGorelli** [comment]
> looks like `pep8speaks` doesn't respect `noqa`...

any objections to removing `pep8speaks`? We already fail CI if `flake8` fails, so I'm not sure what value it adds

File: pandas/core/array_algos/take.py [hunk 1]
```diff
@@ -26,26 +26,12 @@
...
-    indexer,
-    axis: int = ...,
-    fill_value=...,
-    allow_fill: bool = ...,
-) -> np.ndarray:
-    ...
-
-
+def take_nd(arr: np.ndarray, indexer, axis: int = ..., fill_value=..., allow_fill: bool = ...) -> np.ndarray: ...  # noqa
 @overload
```
Comment (jbrockmendel) [inline pandas/core/array_algos/take.py:32]: can the noqa be made specific, i think `# noqa:F501,E704`

Comment (MarcoGorelli) [reply pandas/core/array_algos/take.py:32]: done, thanks!

**Comment by simonjayhawkins** [comment]
I think I prefer the status quo

**Comment by MarcoGorelli** [comment]
Sure, let's stick to that then :+1:

**Pull request closed without merging**